ocaml/typing/jkind.ml

```diff
@@ -404,65 +638,29 @@ let fresh_jkind jkind ~why = { jkind; history = Creation why }
 (* constants *)
 
 let any_dummy_jkind =
-  { jkind = Any; history = Creation (Any_creation Dummy_jkind) }
+  { jkind = Jkind_.max; history = Creation (Any_creation Dummy_jkind) }
 
 let value_v1_safety_check =
-  { jkind = Sort Sort.value;
-    history = Creation (Value_creation V1_safety_check)
-  }
+  { jkind = Jkind_.value; history = Creation (Value_creation V1_safety_check) }
 
 let any ~why =
   match why with
   | Dummy_jkind -> any_dummy_jkind (* share this one common case *)
-  | _ -> fresh_jkind Any ~why:(Any_creation why)
+  | _ -> fresh_jkind Jkind_.any ~why:(Any_creation why)
 
-let void ~why = fresh_jkind (Sort Sort.void) ~why:(Void_creation why)
+let void ~why = fresh_jkind Jkind_.void ~why:(Void_creation why)
 
 let value ~(why : value_creation_reason) =
   match why with
   | V1_safety_check -> value_v1_safety_check
-  | _ -> fresh_jkind (Sort Sort.value) ~why:(Value_creation why)
-
-let immediate64 ~why = fresh_jkind Immediate64 ~why:(Immediate64_creation why)
-
-let immediate ~why = fresh_jkind Immediate ~why:(Immediate_creation why)
-
-let float64 ~why = fresh_jkind (Sort Sort.float64) ~why:(Float64_creation why)
-
-type const = Jane_asttypes.const_jkind =
-  | Any
-  | Value
-  | Void
-  | Immediate64
-  | Immediate
-  | Float64
+  | _ -> fresh_jkind Jkind_.value ~why:(Value_creation why)
 
-let string_of_const : const -> _ = function
-  | Any -> "any"
-  | Value -> "value"
-  | Void -> "void"
-  | Immediate64 -> "immediate64"
-  | Immediate -> "immediate"
-  | Float64 -> "float64"
+let immediate64 ~why =
+  fresh_jkind Jkind_.immediate64 ~why:(Immediate64_creation why)
```

Comment:
It's not new code here, but should we be memoizing these?  I don't see any state.

Reply:
(for common creation reasons, I mean.  Maybe not worth doing?)

Reply:
It's a good question. I've added a CR. One challenge is that we're still figuring out what fields we need on the reasons. Alan has added a bunch. So I didn't want to figure this out until that part is done. But when it all settled, we might get a nice little speed boost by doing some memoization.